### PR TITLE
Refactor beat exit

### PR DESCRIPTION
**Pull request by ruflin**
* Introduce Signal function which is called if using CTRL-C or similar
* Run now returns an error and doesn't exist itself anymore
* Fix spooler and crawler shutdown issue

Thanks to @cyrilleverrier for his contribution here.

### CODE REVIEW

**Comment by ruflin** [comment]
@cyrilleverrier have a look

**Comment by cyrilleverrier** [comment]
LGTM

File: libbeat/beat/beat.go [hunk 1]
```diff
@@ -248,27 +249,39 @@ func (b *Beat) Run() error {
...
-// Stop calls the beater Stop action.
-// It can happen that this function is called more then once.
-func (beat *Beat) Stop() {
-	logp.Info("Stopping Beat")
-	beat.BT.Stop()
+var callback sync.Once
+var exit int
+
+func (b *Beat) Signal() {
+	b.error = fmt.Errorf("Beat stop through signal")
```
Comment (ruflin) [inline libbeat/beat/beat.go]: @andrewkroh @urso this is called when the beat exists through CTRL-C. What should the exit code be in your opinion? 1 or 0?

Comment (urso) [inline libbeat/beat/beat.go]: since we use the kill signal to shutdown our beats, I'd assume it to be 0.

Comment (ruflin) [inline libbeat/beat/beat.go]: Ok, will remove the error again in this case.

Comment (andrewkroh) [inline libbeat/beat/beat.go]: I would expect 0 on a clean shutdown triggered by signal.

File: winlogbeat/main.go [hunk 2]
```diff
@@ -9,5 +11,8 @@ import (
 var Name = "winlogbeat"
 
 func main() {
-	beat.Run(Name, "", winlogbeat.New())
+	err := beat.Run(Name, "", winlogbeat.New())
+	if err != nil {
+		os.Exit(1)
+	}
```
Comment (tsg) [inline winlogbeat/main.go]: This should be done in all Beats, right?

**Pull request merged**